src/service/position-observer-impl.js

```diff
@@ -0,0 +1,389 @@
+import {registerServiceBuilderForDoc} from '../service';
+import {viewportForDoc, vsyncFor} from '../services';
+import {getMode} from '../mode';
+import {dev} from '../log';
+import {
+  moveLayoutRect,
+  layoutRectEquals,
+  layoutRectsOverlap,
+} from '../layout-rect';
+import {IframeMessagingClient} from '../../3p/iframe-messaging-client';
+import {MessageType} from '../../src/3p-frame-messaging';
+
+
+/** @const @private */
+const TAG = 'POSITION_OBSERVER';
+
+/**
+ * The positionObserver returned position value
+ *
+ * @typedef {{
+ *  positionRect: ?../layout-rect.LayoutRectDef,
```

Comment:
Please document if these are layout rects, client rects or relative in some other way. As I mentioned before, I have some concerns about these being layout rects, b/c: (a) layout is always relative and we can run into issues of understanding such relativity; (b) it might be too much info to give to an embed.